lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/evaluator/EvaluatorManagerFactory.java

```diff
@@ -90,10 +94,41 @@ private EvaluatorManager getNewEvaluatorManagerInstance(final String id, final E
    */
   public EvaluatorManager getNewEvaluatorManagerForNewlyAllocatedEvaluator(
       final ResourceAllocationEvent resourceAllocationEvent) {
-    final NodeDescriptor nodeDescriptor = this.resourceCatalog.getNode(resourceAllocationEvent.getNodeId());
+    NodeDescriptor nodeDescriptor = this.resourceCatalog.getNode(resourceAllocationEvent.getNodeId());
 
     if (nodeDescriptor == null) {
-      throw new RuntimeException("Unknown resource: " + resourceAllocationEvent.getNodeId());
+      LOG.log(Level.WARNING, "Node descriptor not found for node {0}", resourceAllocationEvent.getNodeId());
+
+      // HOT FIX for YARN with FEDERATION
+      // See JIRA REEF-568
+      // Should be removed once the YARN-2915 is fixed
+      if (resourceAllocationEvent.getRackName().isPresent()) {
+        final String federationAsStr = resourceAllocationEvent.getRackName().get();
+        final boolean federation = Boolean.valueOf(federationAsStr);
+        // if a true value came here, means that we are using federation
+        if (federation) {
```

Comment:
Wouldn't we be able to know whether we are using this federation-fix based on configuration data? Why do we need to extract that knowledge from the rackname? And: What if we wanted to actually record a rack name?